### ISSUE TITLE
PlexShares with 2 versions of a file will not play

### ISSUE THREAD

**Issue by TheGuru276** [umbrellaplug/umbrellaplug.github.io#253]
### Bug

If a Plexshare has 2 copies of a file for example a mkv version and an mp4 version or a AAC version and a DD+ version, if using the composite addon on its own it prompts you to select which version you wish to play. When using the umbrella/Cocoscrapers feature for PlexShares to find sources and select such a source the prompt is not displayed and the process seems to stall.

### Expected Result

For the prompt to be displayed or a choice automatically made. Ideally to see the prompt.

### Steps To Reproduce

Enable PlexShares in Cocoscrapers and add at least 1 PlexShare.
Install and enable the composite addon as the workaround for the 15 minute Plex idle timeout.
Use umbrella to scrape for sources of a tv show episode that the plexshare has multiple versions of that episode.
Attempt to play that episode.

### Kodi Version, Type of Device, Logs, Screenshots, and Additional Info

Kodi 20.1
Android Mecool KM6 Deluxe

Please let me know if my logs would be of any help but I suspect nothing about the prompt is logged and it will require additional logging to capture what is being returned.

### Checklist

- [X] I have included the version of Kodi I'm using and Device in my report.
- [X] I have made a separate issue for each bug that I am reporting.
- [X] I have verified this is not a setting.

**Comment by TheGuru276** [comment]
In preparation for you actually needing logs I have tried to reproduce this issue from a clean boot up but found nothing in the logs due to the fact Kodi locks up and freezes the moment you try to play the source so nothing of value gets written to the logs in time.

**Comment by umbrellaplug** [comment]
Plex is being removed. The next update of umbrella will no longer include the ability to use Plex shares. I'm getting way too many issues opened for Plex related stuff. It's being removed.

**Comment by TheGuru276** [comment]
That's unfortunate, but understandable.

As it is a useful feature that many people are using would you consider leaving it in but with a disclaimer along the lines of "development has reached its conclusion and no further updates or bug fixes shall be addressed"

**Comment by umbrellaplug** [comment]
I'm having trouble understanding the issue here. I don't use composite. Try this version of cocoscrapers and see if it's better.

https://www.dropbox.com/s/vuz09zdkadyg7yq/script.module.cocoscrapers-1.0.11.zip?dl=0

**Comment by TheGuru276** [comment]
Thank you. I will test the new scraper and let you know. In the meantime this is the type of selection screen that probably should pop up when using umbrella/cocoscrapers if there are 2 versions of the file.  
![IMG_20230402_142104](https://user-images.githubusercontent.com/88575969/229331510-505d10d5-9864-47a6-a6e3-aaf4cafee0f4.jpg)

**Comment by TheGuru276** [comment]
No change with the new scraper. I will have to try and redo logs for you as I had been running logs since I logged this and they are larger than necessary but what I believe is the issue/solution is that when the composite addon is passed the request from umbrella/Cocoscrapers to play the file, if there are 2 files to choose from composite pops up the screenshot I attached previous. This is when Kodi stalls/freezers as the popup has trouble being presented correctly. Not even sure if you can find a solution for this as I guess it might be a composite process at this time of which your scraper has little/no control over.

**Comment by umbrellaplug** [comment]
With the latest version I linked here you shouldn't be getting one link for two files. Composite shouldn't need the popup anymore because the different files are now broken out as different results. Please clear your provider cache and try again and let me know. If a server has two versions of a movie like 1080 vs 4k you should see two links for it now instead of 1.

**Comment by TheGuru276** [comment]
Ok great. I'll clear the providers cache and test again. As per the screenshot, should it also list 2 links if there are 2 1080p versions

**Comment by umbrellaplug** [comment]
Each individual media source from each server should now present as it's own result.

**Comment by TheGuru276** [comment]
Ok. I cleared the providers cache and used the example episode in my previous screenshot and on the initial scrape the plexshare in question returned no results. I did a default rescrape all providers and got the link for the 2nd file, the one from cakes, which is the one that has always shown from memory. When I tried to play it Kodi still froze. Would you like my log files from this test?

**Comment by TheGuru276** [comment]
Ive done some further testing and found no issues with movies. Tv shows however had mixed results. Shows such as Ted Lasso, Shrinking returned both the 4k and 1080 links. However I could not get it to show any links for The Mandalorian and I tested a number of episodes from 2 seasons (4k and 1080 are readily visible in my Plex app). And shows like The Good Doctor and Quantum Leap (2022) that have 2 1080 files, show only 1 link but still cause Kodi to freeze.  Unless you want to spend more time to sort it out, your changes are probably good enough.

**Comment by TheGuru276** [comment]
``
[kodi.log](https://github.com/umbrellaplug/umbrellaplug.github.io/files/11132148/kodi.log)
[umbrella.log](https://github.com/umbrellaplug/umbrellaplug.github.io/files/11132149/umbrella.log)
[cocoscrapers.log](https://github.com/umbrellaplug/umbrellaplug.github.io/files/11132150/cocoscrapers.log)

**Comment by umbrellaplug** [comment]
Looks to be something with composite. From the logs I can see the sources being passed to composite by the structure of the link. There's just not much more I can do.

**Comment by TheGuru276** [comment]
Thank you for making the changes that you have done

**Comment by TheGuru276** [comment]
I don't expect you to spend any more time on this but I just wanted to make you aware of some additional findings.  It's the additional installation of the composite add-on that seems to be causing differences in results. I've had to disable the composite addon to confirm if your latest changes have fixed the issue with the non plex.direct server that user dabutcher has been assisting you with as we both have access to that PlexShare. It only works for me with composite disabled. If I have composite enabled the links still do not play. While also having composite disabled the duel 1080p links from other servers started showing up as separate links. As soon as I clear the provider cache and re-enable the composite addon only one link shows up. This seems weird behaviour as I wouldn't have thought the use or installation of composite had anything to do with the scraping for links, not until you actually tried to start playback.
